### ISSUE TITLE
Could not load type 'QRCoder.QRCode' from assembly 'QRCoder, Version=1.4.3.0'

### ISSUE THREAD

**Issue by EvanFos** [BrandonPotter/GoogleAuthenticator#94]
After updating to GoogleAuthenticator v2.4.0 (#90) via Nuget the following error is thrown in .NET 6.0 when attempting to generate a QR Code URL using QRCoder v1.4.3.0.

TypeLoadException: Could not load type 'QRCoder.QRCode' from assembly 'QRCoder, Version=1.4.3.0, Culture=neutral, PublicKeyToken=c4ed5b9ae8358a28'.

This error does not occur with Googleauthenticatior 2.3.0 which included the QRCoder v1.4.2.0 dependency.

**Comment by ahwm** [comment]
Ah interesting. I suspect the issue is that due to #69 .NET 6 isn’t actually supported yet, unfortunately. I’ll take a look and see if there might be some workarounds.

Are there any other details you can provide such as which platform you’re on?

**Comment by EvanFos** [comment]
The platform would probably help, sorry about that! The app referencing GoogleAuthenticator is a .NET 6.0 Razor pages app running on Windows 10. for dev and Windows Server/IIS for deployment.

For what it's worth, I was also using the previous version of GoogleAuthenticator/QRCoder on .NET 6 with no issues until updating the package. Maybe it was just a fluke that it actually happened to work on .NET 6 before the package update?

Please let me know if any other details would be helpful.

Thanks!

**Comment by klanglie** [comment]
I had the same problem. I am using Windows targeting net6.0 (AspNetCore) I decided to reference the source code to this project directly to work around the issue. The fix, which will also allow this to work on non-windows platforms, was to change GenerateQrCodeUrl to:

```
                using (var qrGen = new QRCodeGenerator())
                using (var qrCode = qrGen.CreateQrCode(provisionUrl, QRCodeGenerator.ECCLevel.Q))
                using (var qrBmp = new BitmapByteQRCode(qrCode))
                {
                    qrCodeUrl = $"data:image/png;base64,{Convert.ToBase64String(qrBmp.GetGraphic(qrPixelsPerModule))}";
                }
```

OLD:
```
                //using (var qrGenerator = new QRCodeGenerator())
                //using (var qrCodeData = qrGenerator.CreateQrCode(provisionUrl, QRCodeGenerator.ECCLevel.Q))
                //using (var qrCode = new QRCode(qrCodeData))
                //using (var qrCodeImage = qrCode.GetGraphic(qrPixelsPerModule))
                //using (var ms = new MemoryStream())
                //{
                //    qrCodeImage.Save(ms, System.Drawing.Imaging.ImageFormat.Png);
                //    qrCodeUrl = $"data:image/png;base64,{Convert.ToBase64String(ms.ToArray())}";
                //}
```

**Comment by FighterLuo** [comment]
> I had the same problem. I am using Windows targeting net6.0 (AspNetCore) I decided to reference the source code to this project directly to work around the issue. The fix, which will also allow this to work on non-windows platforms, was to change GenerateQrCodeUrl to:
> 
> ```
>                 using (var qrGen = new QRCodeGenerator())
>                 using (var qrCode = qrGen.CreateQrCode(provisionUrl, QRCodeGenerator.ECCLevel.Q))
>                 using (var qrBmp = new BitmapByteQRCode(qrCode))
>                 {
>                     qrCodeUrl = $"data:image/png;base64,{Convert.ToBase64String(qrBmp.GetGraphic(qrPixelsPerModule))}";
>                 }
> ```
> 
> OLD:
> 
> ```
>                 //using (var qrGenerator = new QRCodeGenerator())
>                 //using (var qrCodeData = qrGenerator.CreateQrCode(provisionUrl, QRCodeGenerator.ECCLevel.Q))
>                 //using (var qrCode = new QRCode(qrCodeData))
>                 //using (var qrCodeImage = qrCode.GetGraphic(qrPixelsPerModule))
>                 //using (var ms = new MemoryStream())
>                 //{
>                 //    qrCodeImage.Save(ms, System.Drawing.Imaging.ImageFormat.Png);
>                 //    qrCodeUrl = $"data:image/png;base64,{Convert.ToBase64String(ms.ToArray())}";
>                 //}
> ```

good, It works, thanks

**Comment by flytzen** [comment]
Fixed in #102.
Will be released in 2.5. See #104
